README.md

```diff
@@ -80,11 +78,45 @@ resource_group_name = ENV['RESOURCE_GROUP_NAME']
 port = ENV['MSI_PORT'] || 50342 # If not provided then we assume the default port
 ```
 
-Now, we will create token credential using `MSITokenProvider`.
+<a id="msi-sa"></a>
+### Create a System Assigned MSI Token Provider
+
+We can create token credential using `MSITokenProvider` with System Assigned Identity:
+
+```ruby
+# Create System Assigned MSI token provider
+provider = MsRestAzure::MSITokenProvider.new
+```
+
+<a id="msi-ua"></a>
+### Create a User Assigned MSI Token Provider
+To create a User Assigned Identity, you would need to provide a reference to your User Assigned object in order to create an instance. You can provide a
+`client_id`, an `object_id` (Active Directory IDs) or the MSI resource id that must conform to:
+ `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/msiname`
 
+The fastest way to get a `client_id` is to use the CLI 2.0: `az identity show -g myR -n myMSi`
+
+You can get the `object_id` using the `az ad sp show --id <client_id>` command, or througth the Azure Portal in the Active Directory section.
+
+You can also use the `azure_mgmt_msi` package.
+
+```ruby
+    # Create User Assigned MSI token provider using client_id
+    provider = MsRestAzure::MSITokenProvider.new(port, settings,  {:client_id => '00000000-0000-0000-0000-000000000000' })
```

Comment:
Is port required if you use User Assigned?

Reply:
yes, even though port and settings are optional parameter, in order to pass the 3rd parameter, I need to provide the other 2. I would have changed the signature of the method to be a hash for all the arguments, but didn't want to cause a breaking change.